cogs/verification.py

```
@@ -1,67 +1,381 @@
-import discord
-from discord import app_commands
-from discord.ext import commands
+import string
+import random
 import os
 import logging
 from zlib import adler32
+from dataclasses import dataclass
+
+import discord
+from discord import app_commands, ui
+from discord.ext import commands
+from mailersend import emails
+import jwt
+
+
+@dataclass
+class UserInfo:
+    name: str
+    username: str
+    email: str
+    zid: str
+
+
+class VerifyModalUNSW(ui.Modal):
+    """Modal for UNSW students."""
+
+    def __init__(self, db):
+        super().__init__(title="UNSW Verification")
+        self.db = db
+
+        # Required fields for UNSW
+        self.first_name = ui.TextInput(label="First Name")
+        self.last_name = ui.TextInput(label="Last Name")
+        self.zid = ui.TextInput(label="zID (e.g. z1234567)")
+
+        # Add to the modal
+        self.add_item(self.first_name)
+        self.add_item(self.last_name)
+        self.add_item(self.zid)
+
+    def fix_zid(self, zid):
+        if zid.startswith("z".casefold()):
+            return
+        return "z" + zid
```

- "z".casefold() does nothing, as "z" is already lowercase. I assume you want to make the zid lowercase and to do that you can do
```python
zid = zid.lower()
```
- Note we must reject any bad zids. To do this you can throw an error with a good error message, and in the modal's on_error() function we can give feedback to the user. It is not good practice to leave error handling only in the backend as then we can't tell users why they're skill issued.

---

not sure how to throw an error message without closing the modal